[PATCH 2/7] [clang] improve NestedNameSpecifier: misc small clang changes

mizvekov opened the pull request:
Other changes

Second patch in the series starting at https://github.com/llvm/llvm-project/pull/147835

mizvekov committed:
[PATCH 2/7] [clang]  improve NestedNameSpecifier: misc small clang changes

Other changes

Second patch in the series starting at #147835

DeinAlptraum reviewed (commented):
Re: Python bindings changes.
The tests currently fail because the libclang.so cannot even be built. Is this expected, as in, can it only be built after the last patch again? (I see that the tests passed on patch 6/6).

This should also probably get an entry in the bindings' breaking changes section of the release notes.

erichkeane reviewed (approved):
I did as in depth of a review of this as I could, nothing sticks out to me.

efriedma-quic commented on clang/bindings/python/tests/cindex/test_type.py:
```diff
@@ -63,7 +63,7 @@ def test_a_struct(self):
         self.assertEqual(fields[1].spelling, "b")
         self.assertFalse(fields[1].type.is_const_qualified())
-        self.assertEqual(fields[1].type.kind, TypeKind.ELABORATED)
+        self.assertEqual(fields[1].type.kind, TypeKind.TYPEDEF)
```
Unlike most LLVM APIs, we try to keep the libclang API as stable as possible across clang versions... so we need to continue reserving the number 119.  Given that, there's not really any compelling reason to delete the enum itself.

(This is a breaking change in the sense that AST that would previously have contained an ELABORATED type no longer contain such a type, but the breakage from that should be limited, and there isn't really anything we can do if the underlying structure of the type sugar changes.)

mizvekov replied:
Alright, I'll leave the enum for now, we will have until clang 22 ships in order to revisit this.

mizvekov committed:
[PATCH 3/7] [clang] improve NestedNameSpecifier: test changes (#148014)

clang test changes

Third patch in the series, starting at
https://github.com/llvm/llvm-project/pull/147835
